processor/src/test/java/org/mapstruct/ap/test/conversion/url/URLConversionTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.conversion.url;
+
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Tests conversions between {@link java.net.URL} and String.
+ *
+ * @author Adam Szatyin
+ */
+@WithClasses({ Source.class, Target.class, URLMapper.class })
+public class URLConversionTest {
+
+    @ProcessorTest
+    public void shouldApplyURLConversion() throws MalformedURLException {
+        Source source = new Source();
+        source.setURL( new URL("https://mapstruct.org/") );
+
+        Target target = URLMapper.INSTANCE.sourceToTarget( source );
+
+        assertThat( target ).isNotNull();
+        assertThat( target.getURL() ).isEqualTo( source.getURL().toString() );
+    }
+
+    @ProcessorTest
+    public void shouldApplyReverseURLConversion() throws MalformedURLException{
+        Target target = new Target();
+        target.setURL( "https://mapstruct.org/" );
+
+        Source source = URLMapper.INSTANCE.targetToSource( target );
+
+        assertThat( source ).isNotNull();
+        assertThat( source.getURL() ).isEqualTo( new URL( target.getURL() ) );
+    }
+
+    @ProcessorTest
+    public void shouldHandleInvalidURLString() {
```

Comment:
This is a great test. Let's add one more test where the mapping method throws `MalformedURLException` as well.